### ISSUE TITLE
RouteLineResources.destinationWaypointIcon - iconKeepUpright

### ISSUE THREAD

**Issue by Christophe668** [mapbox/mapbox-navigation-android#5004]
**Mapbox Navigation SDK version: 2.0-rc.7**

### Steps to trigger behavior
```
.originWaypointIcon(R.drawable.cby_ic_map_start)
.destinationWaypointIcon(R.drawable.cby_ic_map_end)
```

### Expected behavior
![Screenshot_20211019-142448](https://user-images.githubusercontent.com/1552554/137909218-43d57c12-83a4-4460-894e-d4eca7e933f1.jpg)

### Actual behavior
![Screenshot_20211019-142034](https://user-images.githubusercontent.com/1552554/137909234-f8187c4a-88c6-4057-92ac-9e2618c3c8a5.jpg)

Would it be possible to have to opportunity to configure the start/end points? 
In this case, we would have to have the possibility to use keep `iconKeepUpright`

What do you suggest? To draw it ourselves or have to possibility to do it in the SDK directly?

Thanks

**Comment by harikhalsa** [comment]
@cafesilencio I'm still experiencing this issue in the newest versions of the SDK.
![Screenshot_1642544946](https://user-images.githubusercontent.com/32640267/150029463-53f93f2d-f86b-4ef4-b6da-5f59f8d5b20f.png)
![Screenshot_1642545015](https://user-images.githubusercontent.com/32640267/150029472-ccf86056-1d0e-4eb5-9218-90f3ec2c1b21.png)

**Comment by cafesilencio** [comment]
The `iconKeepUpright=true` was added in Oct. of 2021.  The modification addressed in this issue was to expose options for the placement of the icon via `iconOffset` and `iconAnchor`.  From the original comment above:

>Would it be possible to have to opportunity to **configure the start/end points**?
In this case, we would have to have the possibility to use keep iconKeepUpright

The change was first included in release [2.3.0-beta.1](https://github.com/mapbox/mapbox-navigation-android/releases/tag/v2.3.0-beta.1)

In the screen shots above it appears to me the icon is still upright but angled.  I did some testing and it looks like this can be influenced with the `iconPitchAlignment` option which is set to `IconPitchAlignment.MAP` in the Navigation SDK. This setting goes back to at least version 2.0 and probably 1.x as well.  This option isn't currently exposed to users of the SDK but it could be.

I tested using `IcontPitchAlignment.VIEWPORT` for the `iconPitchAlignment` option and I see a difference.  In the image below is a view of a pin looking straight down on the map followed by an image with the default `iconPitchAlignment`= `IconPitchAlignment.MAP` during navigation which pitches the map, followed by an image with `iconPitchAlignment`= `IconPitchAlignment.VIEWPORT`.  Based on the images I think what is seen in the screen shots above is correct according to the configuration.  We can expose the `iconPitchAlignment` option externally for a future release if that is desired.

![pins](https://user-images.githubusercontent.com/2249818/156479629-d69c9271-0ce9-45f8-ba9d-5c453bad832d.png)
